src/NServiceBus.Core/OpenTelemetry/Metrics/ReceiveDiagnosticsBehavior.cs

```diff
@@ -18,12 +19,14 @@ public ReceiveDiagnosticsBehavior(string queueNameBase, string discriminator)
     public async Task Invoke(IIncomingPhysicalMessageContext context, Func<IIncomingPhysicalMessageContext, Task> next)
     {
         context.MessageHeaders.TryGetValue(Headers.EnclosedMessageTypes, out var messageTypes);
+        var messageTypeHeader = !string.IsNullOrEmpty(messageTypes) ? messageTypes.Split(';').FirstOrDefault() : default;
```

Comment:
Would we ever want to handle a weirdly formatted message type string that starts with a `;`.  Maybe this is too edge and we'd want to fix the thing creating the buggy data instead.

```csharp
using System;
using System.Linq;

namespace HelloWorld
{
	public class Program
	{
		public static void Main(string[] args)
		{
		  var messageTypes = ";ClassName1,unecessary information;ClassName2,unecessary information";
			var messageTypeHeader = !string.IsNullOrEmpty(messageTypes) ? messageTypes.Split(';').FirstOrDefault(s => !string.IsNullOrEmpty(s)) : null;
			var messageTypeName = !string.IsNullOrEmpty(messageTypeHeader) ? messageTypeHeader.Split(',').FirstOrDefault(s => !string.IsNullOrEmpty(s)) : null;
			
			Console.WriteLine(messageTypeName);
		}
	}
}
```

Reply:
@lailabougria Usually, we like to cache these types of string manipulations in a ConcurrentDictionary for performance reasons?

https://github.com/Particular/NServiceBus/blob/fd29208107e147251c10c1663b4c62c0d1e110c4/src/NServiceBus.Core/Pipeline/Incoming/DeserializeMessageConnector.cs#L72

https://github.com/Eventellect/SupportRepro/blob/39b6264282688f21737990014e2b5b72faa4b660/src/NSB.Extensions/EnclosedMessageTypesMappingBehavior.cs#L59